suricata/update/rule.py

```diff
@@ -118,7 +118,11 @@ def id(self):
         :returns: A tuple (gid, sid) representing the ID of the rule
         :rtype: A tuple of 2 ints
         """
-        return (int(self.gid), int(self.sid))
+        try:
+            return (int(self.gid), int(self.sid))
+        except TypeError:
+            print("Error : Sid cannot be null")
```

Comment:
It would probably make more sense to use logger to log this error. Also, since we do not know whether sid or gid failed, we should make the error message clearer.